File: core/src/main/java/com/cc/core/command/Messenger.java
package com.cc.core.command;

import com.cc.core.WorkerHandler;
import com.cc.core.actions.Actions;
import com.cc.core.utils.Utils;

public class Messenger {
    public static void sendCommand(final Command cmd, final Callback callback) {
        if (cmd == null) {
            callback.onResult(null);
            return;
        }

        WorkerHandler.postOnWorkThread(new Runnable() {
            @Override
            public void run() {
                String result;
                if (Utils.isEmpty(cmd.getArgs())) {

                    result = Actions.Companion.executeCommand(cmd.getKey());
                } else {
                    result = Actions.Companion.executeCommand(cmd.getKey(), cmd.getArgs().toArray());
                }
                callback.onResult(result);
            }
        });

    }
}

File: core/src/main/java/com/cc/core/command/Messenger.kt
package com.cc.core.command

import com.cc.core.WorkerHandler
import com.cc.core.actions.Actions
import com.cc.core.utils.Utils

class Messenger {
    companion object {
        fun sendCommand(cmd: Command?, callback: Callback?) {
            if (cmd == null) {
                callback!!.onResult(null)
                return
            }

            WorkerHandler.postOnWorkThread {
                val result: String = if (Utils.isEmpty(cmd.getArgs())) {

                    Actions.executeCommand(cmd.getId(), cmd.getKey())
                } else {
                    Actions.executeCommand(cmd.getId(), cmd.getKey(), *cmd.getArgs()!!.toTypedArray())
                }
                callback!!.onResult(result)

            }

        }
    }
}